src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.Unix.cs

```diff
@@ -188,5 +188,16 @@ public static unsafe void FreeBSTR(IntPtr ptr)
                 Interop.Sys.MemFree(ptr - sizeof(IntPtr));
             }
         }
+
+        internal static Type? GetTypeFromProgID(string progID, string? server, bool throwOnError)
+        {
+            if (progID == null)
+                throw new ArgumentNullException(nameof(progID));
+
+            if (throwOnError)
+                throw new PlatformNotSupportedException(SR.PlatformNotSupported_ComInterop);
+
+            return null;
```

Comment:
We will return `null` instead of throwing when `throwOnError` is false. It matches what we do in other similar situations.